packages/mobile/src/account/profileInfo.ts

```diff
@@ -90,31 +91,42 @@ export function* uploadNameAndPicture() {
 }
 
 // this function gives permission to the recipient to view the user's profile info
-export function* giveProfileAccess(recipientAddresses: string[]) {
+export function* giveProfileAccess(recipientAddress: string) {
+  // TODO: check if key for recipient already exists, skip if yes
   try {
-    // TODO: check if key for recipient already exists, skip if yes
+    const walletToAccountAddress: WalletToAccountAddressType = yield select(
+      walletToAccountAddressSelector
+    )
+    const accountAddress = walletToAccountAddress[recipientAddress]
+    if (!accountAddress) {
+      Logger.info(TAG, 'cannot find account address for wallet address', recipientAddress)
```

Comment:
If we don't have it should we try to give anyways to the `accountAddress` in case it's the same?